Test/git-issues/git-issue-930.dfy

```diff
@@ -0,0 +1,139 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+datatype Result<T> = Success(value: T) | Failure(error: string)
+{
+  predicate method IsFailure() {
+    Failure?
+  }
+  function method PropagateFailure<U>(): Result<U>
+    requires Failure?
+  {
+    Failure(this.error)
+  }
+  function method Extract(): T
+    requires Success?
+  {
+    value
+  }
+}
+
+class ClassA<T> {
+  const s: seq<T>
+
+  constructor (s: seq<T>) {
+    this.s := s;
+  }
+
+  method Read() returns (res: Result<seq<T>>)
+    ensures res.Success? ==> 1 <= |res.value|
+  {
+    if 1 <= |s| {
+      if * {
+        return Success(s);
+      }
+    }
+    return Failure("failing");
+  }
+
+  method ReadX<U(==)>(x: U, y: U) returns (res: Result<seq<U>>)
+    ensures res.Success? ==> 1 <= |res.value|
+  {
+    if x != y {
+      return Success([x, y]);
+    }
+    return Failure("failing");
+  }
+}
+
+// Methods M0 and M1 are the same, except that M0 gives the type of
+// "numbers", whereas M1 leaves that type to be inferred. In both
+// cases, it should be known that a.Read() returns a Result<seq<int>>,
+// but type inference fails to instantiate the T in ClassA<T>, so
+// a.Read() is typed as returning a Result<seq<T>>, which is bogus.
+
+method M0() returns (res: Result<int>) {
+  var a: ClassA<int> := new ClassA([57, 58, 59]);
+  var numbers: seq<int> :- a.Read();  // bogus: says expected Result<seq<int>>, but got Result<seq<T>>
```

Comment:
Please change the various occurrences of "bogus" in this test file to something along the lines of

    // regression test: this had once said ...

Reply:
Fixed